system-tests/__snapshots__/vite_dev_server_fresh_spec.ts.js

```diff
@@ -391,11 +395,15 @@ We dynamically generated a new test to display this failure.
   1 failing
 
   1) is missing React:
-     ReferenceError: The following error originated from your test code, not from Cypress.
+     ReferenceError: The following error originated from your application code, not from Cypress.
```

Comment:
Weird one - I think application is more accurate here. In this example the user wrote a JSX file but didn't do `import React from 'react'` - I'd say that's their application code.

For CT, application/spec is really a blurred line - it's one bundle and one iframe, so it's kind of the same thing.

Reply:
Yeah a bit weird, I added a `setTimeout(() => throw new Error('wtf))` inside an `it` and the error is the same (type=application). Not sure of a way to differentiate between app and test code when it comes to CT unless you looked at the error stack.

I would be in favor of it always saying `test` code for CT

Reply:
We could do this, we'd need to move this conditional error listener. The main problem is we will never be "correct" unless we do some stack trace introspection, which I think is going to really blow out the scope. If you move your `setTimeout` error to your application code, then "test code" would be incorrect.

 Maybe we can make this more generic for CT - something like "The following error originated from your code"? The main message we are conveying here is "look at the code YOU wrote for the problem". I'll see if I can make this more generic.

Reply:
Actually, I thought about it more - this is probably worth spending a bit of time on, it's quite a foundational part of Cypress. I am not sure if there is a good way to know if it's spec or app error, though. I'll take a look.

Reply:
Ok, something like

1. Get stack trace
2. Get origin of error
3. If spec file -> spec error
4. else -> App error

This means we need to see if the file in the stack traces matches the specPattern glob. This is tricky.

Reply:
Yeah I agree, maybe I'll spend 1 more hour or so. Basic heuristic could be

- spec file OR support file: test code
- anything else: application code

If this is too much work, I'll just pick one and go with it - either way, the stack trace shows, which is the main thing.

Reply:
@ZachJW34 I thought about this more. I compiled a list of every error and how the UI looks after this change: https://www.notion.so/cypressdx/Errors-adf3361e7c174c388404142148a865a5

WDYT? "Application Error" seems more accurate that "Spec Error" for the majority of cases.